tools/test/travis-ci/doxy-spellchecker/spell.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash -eu
+# mbed Microcontroller Library
+# Copyright (c) 2018 ARM Limited
+# SPDX-License-Identifier: Apache-2.0
+
+set -o pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+ERRORS=0
+
+# Loops use here strings to allow them to run in the main shell and modify the correct version of
+# the error counter global variable
+while read file; do
+    echo "${file}"
+    res=$(awk '/\/\*\*/,/\*\//' "${file}" | cut -d '/' -f2 | sed 's/0x[^ ]*//' | sed 's/[0-9]*//g')
+
+    # Select a token to begin on, then a formating option such as strip all text between the start
+    # and end token, strip an entire line containing the start token, or strip a portion of a line
+    # containing the start token. Select an appropiate end token. The tokens and formats are index
+    # matched.
+    start_tokens=(  "/@code"
+                    "/addtogroup"
+                    "defgroup"
+                    "<"
+                    "()"
+                 )
+
+    formats=(   'strip_between'
+                'strip_between'
+                'strip_line'
+                'strip_between_sameline'
+                'strip_token'
+            )
+
+    end_tokens=(    "/@endcode"
+                    "/\*"
+                    ""
+                    ">"
+                    ""
+               )
+
+    # Stripping strings between tokens P1-P2 and P3-P4 inclusively ran into issues depending
+    # on if the tokens were on the same line or not.
+    #_________________________________________
+    # Don't remove this P1 remove me P2
+    # Keep me
+    # P3
+    #   Remove me too please
+    # P4
+    # Keep me too
+    # Still here P1 But this shouldn't be P2
+    #_________________________________________
+    #
+    # Opted for having two separate formats. In particular this formatting issue came up when
+    # trying to strip the code segments and template type arguments between '<, >' as the multiline
+    # sed command would strip the entire line, causing the removal string to span across the entire file
+    # when trying to match the next end token (above format when stripping everything between P1 and P2
+    # would end up with just "Don't remove this" and the rest of the file stripped).
+
+    for ((i=0;i<${#start_tokens[@]};++i)); do
+        filter=""
+        if [[ "${formats[i]}" == 'strip_between' ]]; then
+            filter=$(echo "${res}" | sed ""${start_tokens[i]}"/,"${end_tokens[i]}"/d")
```

Comment:
```suggestion
            filter=$(<<< "${res}" sed ""${start_tokens[i]}"/,"${end_tokens[i]}"/d")
```

This and other instances can be replaced with here-strings.
This and other instance